### DISCUSSION TITLE
Configuring MONAILabel and CVAT running in separate docker?

### DISCUSSION THREAD

**Discussion by bilalcodehub** [Project-MONAI/MONAILabel#1281] [General]
Hi All,

MONAILabel and CVAT versions: Latest

I want to use MONAILabel for surgical video annotation. I can only install the software in docker. I installed MONAILabel docker and checked it through the radiology app using 3D Slicer, so both are working perfectly.

For surgical videos, I need to install CVAT which I did through docker. Now I have two dockers with one running MONAILabel and another CVAT. How can I configure CVAT & MONAILabel in this setting so that they work with each other for video labeling?

I will greatly appreciate your help on this.

Thanks in advance
Bil

**Comment by tangy5** [comment]
You can refer to this page: https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/cvat#installation
The "deploy" script will deploy MONAI Label models to use in CVAT. 

For a full instruction, step-by-step instation of MONAI Label, CVAT, Nuclio sutff, here is a tutorial: https://github.com/Project-MONAI/tutorials/blob/main/monailabel/monailabel_endoscopy_cvat_tooltracking.ipynb

But I feel you already done the CVAT installation, just need the  "deploy" step. 

Hope this helps.

**Reply by bilalcodehub** [reply]
Thanks @tangy5 for the response and for sharing such a detailed tutorial. Much appreciated.

When I run this `./deploy` command, I get the error saying no such file or directory. Please check the screenshot below:

<img width="488" alt="image" src="https://user-images.githubusercontent.com/7869659/217101109-1e00566f-4678-45f0-a6da-5d7f4516db6f.png">

Any idea why this might be the issue?

**Reply by tangy5** [reply]
Hi @bilalcodehub , the file is not a part of CVAT. It's a "plugin" developed by MONAI Label team. 
The file is here: https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/cvat/endoscopy

You can read the tutorial or the Github page for more details, such as how to get them.

**Reply by bilalcodehub** [reply]
Hi @tangy5: I see these files are copied from `monailabel` plugins folder to the `workspace` folder created on the local computer with following command:
`docker cp $(docker container ls | grep 'nvcr.io/nvidia/clara/monai-toolkit' | awk '{print $1}'):/opt/conda/monailabel/plugins/cvat workspace`

In my installation, MONAILabel is running through the docker container, not installed on the local system so can't run the above command as the `docker container ls | grep 'nvcr.io/nvidia/clara/monai-toolkit` part of the command is returning an error when I run it on the docker with MONAILabel. Do I simply grab plugins from MONAILabel GitHub as the tutorial describes in the next step and deploy them to CVAT?

Apologies for naivety as this is the first time I am using MONAILabel and appreciate any guidance on this.

**Reply by tangy5** [reply]
Ok, in this case, you can refer to option 2 in the tutorial, in the docker, you can either pull the file or mount the file to the docker. 

git clone https://github.com/Project-MONAI/MONAILabel.git
Deploy all endoscopy models
./MONAILabel/plugins/cvat/deploy.sh endoscopy

**Reply by bilalcodehub** [reply]
I was able to run the option 1 command and copy the files from docker. The docker name is different on my computer so I needed to adjust the command as follows:
`docker cp $(docker container ls | grep 'projectmonai/monailabel:latest' | awk '{print $1}'):/opt/conda/monailabel/plugins/cvat workspace`
Thanks for the help.
